src/Client.php

```diff
@@ -497,14 +506,53 @@ public function createService(string $name, string $deployment_name, int $port,
       ],
     ];
 
-    return $this->request($resourceMethod['action'], $uri, $service);
+    $annotations = ['app' => $app_name];
+    $this->applyAnnotations($service, $annotations);
+
+    $result = $this->request($resourceMethod['action'], $uri, $service);
+    if ($result && $app_name != $name) {
+      $this->groupService($app_name, $name);
+    }
+
+    return $result;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function updateService(string $name, string $deployment_name, int $port, int $target_port, string $app_name) {
+    $resourceMethod = $this->getResourceMethod(__METHOD__);
+    $uri = $this->createRequestUri($resourceMethod['uri']);
+
+    $service = $this->getService($name);
+
+    $annotations = ['app' => $app_name];
+    $this->applyAnnotations($service, $annotations);
+
+    $result = $this->request($resourceMethod['action'], $uri, $service);
+    if ($result && $app_name != $name) {
+      $this->groupService($app_name, $name);
+    }
+
+    return $result;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function updateService(string $name, array $data) {
-    // TODO: Implement updateService() method.
+  public function groupService(string $app_name, string $name) {
+    $resourceMethod = $this->getResourceMethod(__METHOD__);
+    $uri = $this->createRequestUri($resourceMethod['uri'], [
+      'name' => (string) $app_name,
```

Comment:
Superfluous cast. The method signature enforces string via type hinting.

Reply:
This is a copy pasta, the patten is repeated heaps, fixing the one i did, leaving the rest for later.